orttraining/orttraining/models/runner/training_runner.cc

```diff
@@ -69,6 +69,9 @@ TrainingRunner::TrainingRunner(Parameters params, const Environment& env, Sessio
   if (params.deepspeed_zero.stage != 0)
     ORT_ENFORCE(params.use_nccl,
                 "DeepSpeed ZeRO partitioning is only supported with NCCL distributed training.");
+  if(params.deepspeed_zero.stage > 1)
+    ORT_ENFORCE(params.gradient_accumulation_steps == 1,
+                "DeepSpeed ZeRO stage 2 is only supported with accumulation step == 1.");
```

Comment:
Is this required for functionality?  Or just for performance?

Reply:
Hmm, currently functional wise also only supports the step =1. The change is in zero_optimizer_graph_builder after gradient scaling, if we support accumulation > 1, we need to insert NcclReduce nodes in each iteration of accumulation steps.